include/experimental/__p0009_bits/extents.hpp

```diff
@@ -13,511 +13,612 @@
-  using size_type = make_unsigned_t<index_type>;
+  MDSPAN_INLINE_FUNCTION
+  constexpr T operator[](size_t r) const { return impl_t::get(r); }
+  MDSPAN_INLINE_FUNCTION
+  constexpr static T get(size_t r) { return impl_t::get(r); }
+  template <size_t r> MDSPAN_INLINE_FUNCTION constexpr static T get() {
+    return impl_t::template get<r>();
+  }
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t size() { return sizeof...(Values); }
+};
 
-// internal typedefs which for technical reasons are public
-  using __storage_t = detail::__partially_static_sizes_tagged<detail::__extents_tag, index_type, size_t, Extents...>;
 
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-  _MDSPAN_NO_UNIQUE_ADDRESS __storage_t __storage_;
-#else
-  using __base_t = detail::__no_unique_address_emulation<__storage_t>;
-#endif
+// ------------------------------------------------------------------
+// ------------ index_sequence_scan ---------------------------------
+// ------------------------------------------------------------------
 
-// private members dealing with the way we internally store dynamic extents
- private:
+// index_sequence_scan takes compile time values and provides get(r)
+// and get<r>() which return the sum of the first r-1 values.
 
-  MDSPAN_FORCE_INLINE_FUNCTION _MDSPAN_CONSTEXPR_14
-  __storage_t& __storage() noexcept {
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-    return __storage_;
+// Recursive implementation for get
+template <size_t R, size_t... Values> struct index_sequence_scan_impl;
+
+template <size_t R, size_t FirstVal, size_t... Values>
+struct index_sequence_scan_impl<R, FirstVal, Values...> {
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t get(size_t r) {
+    if (r > R)
+      return FirstVal + index_sequence_scan_impl<R + 1, Values...>::get(r);
+    else
+      return 0;
+  }
+  template <size_t r> MDSPAN_INLINE_FUNCTION constexpr static size_t get() {
+#if MDSPAN_HAS_CXX_17
+    if constexpr (r > R) {
+      return FirstVal + index_sequence_scan_impl<R + 1, Values...>::get(r);
+    } else {
+      return 0;
+    }
 #else
-    return this->__base_t::__ref();
+    return get(r);
 #endif
   }
-  MDSPAN_FORCE_INLINE_FUNCTION
-  constexpr __storage_t const& __storage() const noexcept {
-#if defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS)
-    return __storage_;
+};
+
+template <size_t R, size_t FirstVal>
+struct index_sequence_scan_impl<R, FirstVal> {
+#if defined(__NVCC__) || defined(__NVCOMPILER)
+  // NVCC warns about pointless comparison with 0 for R==0 and r being const
+  // evaluatable and also 0.
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t get(size_t r) {
+    return static_cast<int64_t>(R) > static_cast<int64_t>(r) ? FirstVal : 0;
+  }
+#else
+  MDSPAN_INLINE_FUNCTION
+  constexpr static size_t get(size_t r) { return R > r ? FirstVal : 0; }
+#endif
+  template <size_t r> MDSPAN_INLINE_FUNCTION constexpr static size_t get() {
+#if MDSPAN_HAS_CXX_17
+    if constexpr (r > R) {
```

Comment:
Does this situation actually come up in the code? The return value of `FirstVal` doesn't reflect the stated result of this operation. Wouldn't this only occur on an out of bounds access?

Reply:
this is the wrong way around? shouldn't it be R>r?

Reply:
But as Damien observed above this overload is not used?